### DISCUSSION TITLE
Documentation for the Z3 proof language

### DISCUSSION THREAD

**Discussion by wenkokke** [Z3Prover/z3#5671] [Q&A] [answered]
Is there a specification of Z3 proof language?

**Comment by NikolajBjorner** [comment, marked as answer]
There is a workshop paper, IWIL 2008, that describes the design and main proof rules.
Select rules have been added over time, hyper-resolution.
The proof checker code serves as an entry point for providing an independent description of what is expected,
https://github.com/Z3Prover/z3/blob/master/src/ast/proofs/proof_checker.h
